plugins/gtav/Game.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020-2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "Game.h"
+
+#include "mumble_positional_audio_utils.h"
+
+#include <cstring>
+
+Game::Game(const procid_t id, const std::string &name) : m_proc((id), name) {
+}
+
+Mumble_PositionalDataErrorCode Game::init() {
+	if (!m_proc.isOk()) {
+		return MUMBLE_PDEC_ERROR_TEMP;
+	}
+
+	const Modules &modules = m_proc.modules();
+	const auto iter        = modules.find("GTA5.exe");
+	if (iter == modules.cend()) {
+		return MUMBLE_PDEC_ERROR_TEMP;
+	}
+	// Find out the offset of the memory region
+	const MemoryRegions &regions = iter->second.regions();
+	const auto &r                = regions.cbegin();
+	if (r != regions.cend()) {
+		m_moduleBase = r->address;
+	}
+
+	// Check if we can get meaningful data from it
+	float apos[3], afront[3], atop[3], cpos[3], cfront[3], ctop[3];
+	const char *identity = "";
+	const char *context  = "";
+
+	return fetchPositionalData(apos, afront, atop, cpos, cfront, ctop, &context, &identity);
```

Comment:
This block can be removed if you implement the suggestion below.

Reply:
I quite like that!

Reply:
> This block can be removed if you implement the suggestion below ([#5155 (comment)](https://github.com/mumble-voip/mumble/pull/5155#discussion_r661852945)).
Unfortunately I think there is no other way than to fetch positional data already in the init method, to fix the rapid linking and unlinking when the person is not ingame currently (see https://github.com/mumble-voip/mumble/pull/5155#issuecomment-872094014)

What do you say? I think we will have to add back that part, but can still keep address init in the init method.

Reply:
Another solution could be to pretend to still be able to deliver positional data even if the player is not in-game yet (always sending (0,0,0) as the position) thus preventing the unlink :thinking:

Reply:
> Another solution could be to pretend to still be able to deliver positional data even if the player is not in-game yet (always sending (0,0,0) as the position) thus preventing the unlink

No that wouldn't really work. People would still hear you positonally, but your voice coming from a very weird point of space. (0,0,0) does not disable positional audio, it is just an arbitrary fixed point of space and the only case where that would not behave weirdly is if all Mumble users have (0,0,0) set. In any other case that would not work.


And after all, probing positional data once in init is not such a big of a deal. It would have been a nice extra if that part could have been removed, but that didn't work.